src/partition/repart.c

```diff
@@ -4042,7 +4080,7 @@ static int context_open_copy_block_paths(
                 free_and_replace(p->copy_blocks_path, opened);
 
                 /* When copying from an existing partition copy that partitions UUID if none is configured explicitly */
-                if (sd_id128_is_null(p->new_uuid) && !sd_id128_is_null(uuid))
+                if (!p->new_uuid_is_set && !sd_id128_is_null(uuid))
                         p->new_uuid = uuid;
```

Comment:
I think ` p->new_uuid_is_set` should be set here, even if it does not matter for current logic, just so that we have a self-consistent state.